### ISSUE TITLE
[Bug] 4.5.0.356 - System.NullReferenceException - CarouselView

### ISSUE THREAD

**Issue by johnmackerras** [xamarin/Xamarin.Forms#9761]
### Description
Upgrading to Forms 4.5.0.356 causes a "System.NullReferenceException: 'Object reference not set to an instance of an object.' associated with the carousel view. 

### Steps to Reproduce

1. Open a new Xamarin.Forms app an upgrade Xamarin.Forms to a version which supports Carousel. I used 4.4.0.991640

2. Add a new property ```public List<string> Images { get; set; } = new List<string>();``` to the Models/Item class to represent a list of images.

3. Add this carousel view to ```ItemDetailPage.xaml```
```
        <CarouselView
            x:Name="cvMain"
            Margin="0"
            HeightRequest="200"
            ItemsSource="{Binding Item.Images}"
            EmptyView="No photos to display">
            <CarouselView.ItemTemplate>
                <DataTemplate>
                    <Grid Margin="0" Padding="10">
                        <Frame
                            Margin="0"
                            Padding="0"
                            CornerRadius="5"
                            IsClippedToBounds="True"
                            VerticalOptions="StartAndExpand">
                            <Grid>
                                <Image Aspect="AspectFill" Source="{Binding .}" />
                            </Grid>
                        </Frame>
                    </Grid>
                </DataTemplate>
            </CarouselView.ItemTemplate>
        </CarouselView>
```

4. Set your experamental flags for the CarouselView.

5. Run app and click on any item to move into the Item dteail page. This should work as expected.

6. Update Xamarin.Forms, and re-run. When you click through to the detail page you should now get a "System.NullReferenceException: 'Object reference not set to an instance of an object.'" error.

7. Place a breakpoint on the initlializer ```public ItemDetailPage(ItemDetailViewModel viewModel)``` for ```ItemDetailPage.xaml.cs```. When you step through (using F11) the line at fault seems to be setting the Title string of the BaseViewModel. This does not make sense to me why it's at that particular point or why a string should cause it, but that's as far as I can trace it.

8. Roll back forms to teh previous version and it should work again.

- Version with issue: Xamarin.Forms 4.5.0.356 
- Last known good version:
- IDE: Microsoft Visual Studio Community 2019
Version 16.4.5, .Net croe 3.1

**Comment by hartez** [comment]
[_9761 Repro.zip](https://github.com/xamarin/Xamarin.Forms/files/4278015/_9761.Repro.zip)

**Comment by jsuarezruiz** [comment]
This issue have been fixed by https://github.com/xamarin/Xamarin.Forms/pull/9708

**Comment by rmarinho** [comment]
Hum i need to doublecheck @jsuarezruiz  if this is fixed, the  result seems the same fix, but the cause might be not the same.

**Comment by rmarinho** [comment]
Just confirmed this should be fix on 4.6.0-pre1 (already up), or 4.5.0-sr1 (up next week)
